riptide-core/src/main/java/org/zalando/riptide/DefaultMessageReader.java

```diff
@@ -44,9 +44,13 @@
     private <I> I readBody(final Type type, final ClientHttpResponse response) throws IOException {
         final ResponseExtractor<I> extractor = new HttpMessageConverterExtractor<>(type, converters);
         try {
-            return extractor.extractData(response);
+            try {
+                return extractor.extractData(response);
+            } catch (final IOException | RuntimeException e) {
+                response.close();
```

Comment:
If this fails with its own `IOException` we might loose the original one. Would that be an issue or can we ignore that?

Reply:
Aha! Another reason `SneakyThrows` must go: I've checked the contract of `ClientHttpResponse` which specifies that implementations must swallow `IOExceptions`, hence I went without another `try-catch` and possibly a "suppressed exception". Enter Riptide's `ApacheClientHttpResponse` which actually throws `IOExceptions` :disappointed: 

IMO we should rather change `ApacheClientHttpResponse` to not throw `IOExceptions`. 

You can still have a case for any `RuntimeException` obviously, but in that case I favor swallowing the original one as this is an unexpected scenario.

Reply:
> IMO we should rather change `ApacheClientHttpResponse` to not throw `IOExceptions`.

Sounds good